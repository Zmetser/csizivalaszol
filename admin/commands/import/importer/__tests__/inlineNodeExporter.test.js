/**
 * @flow
 */

const exporter = require('../normalize')

describe.only('line breaks exporter', () => {
  test('should create a paragraph on double line break and remove brs', () => {
    const result = exporter('Na sziasz!<br><br>KÉRDÉS')
    expect(result).toMatchSnapshot()
  })

  test('should remove br from the beginning of the paragraph', () => {
    const result = exporter('Na sziasz!<br><br><br>KÉRDÉS')
    expect(result).toMatchSnapshot()
  })

  test('should preserve single br', () => {
    const result = exporter('Na sziasz!<br>KÉRDÉS<br>asd')
    expect(result).toMatchSnapshot()
  })
})
