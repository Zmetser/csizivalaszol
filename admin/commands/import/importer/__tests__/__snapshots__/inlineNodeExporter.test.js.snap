// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line breaks exporter should create a paragraph on double line break and remove brs 1`] = `
Array [
  Object {
    "type": "Paragraph",
    "value": Array [
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "Na sziasz!",
      },
    ],
  },
  Object {
    "type": "Paragraph",
    "value": Array [
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "KÉRDÉS",
      },
    ],
  },
]
`;

exports[`line breaks exporter should preserve single br 1`] = `
Array [
  Object {
    "type": "Paragraph",
    "value": Array [
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "Na sziasz!",
      },
      Object {
        "type": "LineBreak",
      },
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "KÉRDÉS",
      },
      Object {
        "type": "LineBreak",
      },
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "asd",
      },
    ],
  },
]
`;

exports[`line breaks exporter should remove br from the beginning of the paragraph 1`] = `
Array [
  Object {
    "type": "Paragraph",
    "value": Array [
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "Na sziasz!",
      },
    ],
  },
  Object {
    "type": "Paragraph",
    "value": Array [
      Object {
        "styles": Array [],
        "type": "Text",
        "value": "KÉRDÉS",
      },
    ],
  },
]
`;
